crates/bevy_ui/src/render/ui_material_pipeline.rs

```diff
@@ -407,7 +420,11 @@ pub fn extract_ui_material_nodes<M: UiMaterial>(
 
             // Both vertical and horizontal percentage border values are calculated based on the width of the parent node
             // <https://developer.mozilla.org/en-US/docs/Web/CSS/border-width>
-            let parent_width = uinode.size().x;
+            let parent_width = maybe_parent
```

Comment:
Parent width is also used for `top` and `bottom`, does that make sense?

I would have expected that to use the height

Reply:
This is standard web UI behavior IIRC: this ensures that the borders stay the same size as each other even as the aspect ratio changes. A comment to that effect from an expert would be nice to add though.

@sixfold-origami or @nicoburns can probably confirm.

Reply:
Much of the reason why this is the case now is historical, but I believe the original motivation for doing it this way  avoid circular (/multipass) computations. If vertical padding is a percentage of the parent's height but the parent's height is "auto" then the size of the paddingdepends on the parent's height, but the parent's height also depends on the size of the padding. This problem doesn't occur with width in block layout because auto width in block layout is a "stretch" width and is thus never dependent on the content size (or close enough).

Note that I used padding in this example, because on web it is not possible for a border to be a percentage. That's actually a small extension that only Taffy supports.

Reply:
Thanks for clarifying!
Perhaps an additional comment in the code would be useful (because it does look kinda suspicious), but that doesn't need to be part of this PR